pkg/artifactcache/handler.go

```diff
@@ -386,7 +386,7 @@ func (h *Handler) findCache(db *bolthold.Store, keys []string, version string) (
 
 	for _, prefix := range keys[1:] {
 		found := false
-		if err := db.ForEach(bolthold.Where("Key").Ge(prefix).And("Version").Eq(version).SortBy("Key"), func(v *Cache) error {
+		if err := db.ForEach(bolthold.Where("Key").Ge(prefix).And("Version").Eq(version).SortBy("CreatedAt").Reverse(), func(v *Cache) error {
```

Comment:
Ahh right, I see. It just happened to work for me like this given our use case.

Changed my suggestion to only select keys starting with prefix before ordering by `CreatedAt`. 

The abort cond, now on line 396, is kind of unnecessary now since `RegExp()` in the query will ensure the key has the full prefix.